src/intrinsics/prepack/utils.js

```diff
@@ -86,7 +86,10 @@ export function createAbstract(
     let kind = "__abstract_" + realm.objectCount++; // need not be an object, but must be unique
     result = AbstractValue.createFromTemplate(realm, throwTemplate, type, [locVal], kind);
   } else {
-    let kind = "__abstract_" + nameString; // assume name is unique TODO #1155: check this
+    let kind = "__abstract_" + nameString;
+    if (!realm.isNameStringUnique(nameString)) {
+      throw realm.createErrorThrowCompletion(realm.intrinsics.TypeError, "An abtract value with the same name exists");
```

Comment:
This has been fixed in a [later commit](https://github.com/facebook/prepack/pull/1197/files#diff-1645bddd636bcc8374617f81d0a2d5d5R92)

Apologies for the confusion. I guess I should squash such commits together. I tend to keep certain  commits unsquashed so that the reviewer can see how things changed